### PR TITLE
feat(core): add error server storage service

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/core/src/storage/server-error/server-error.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { DaffError } from '../../errors/public_api';
+import { DaffStorageServiceError } from '../error/error';
+
+/**
+ * An error thrown when there is an attempt to access storage on the server and none is available.
+ */
+export class DaffServerStorageServiceError extends DaffStorageServiceError implements DaffError {
```
Comment (damienwebdev) [inline libs/core/src/storage/server-error/server-error.ts:7]: `DaffServerSideStorageError `

File: libs/core/src/storage/server-error/server-error.service.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+
+import { DaffPersistenceService } from '../persistence.interface';
+import { DaffServerStorageServiceError } from './server-error';
+
+/**
+ * An error storage service meant to be loaded into SSR contexts.
```
Comment (damienwebdev) [inline libs/core/src/storage/server-error/server-error.service.ts:7]:
```
A storage service meant to be loaded into SSR contexts. 
```

File: libs/core/src/storage/server-error/server-error.service.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * It will always throw the {@link DaffServerStorageServiceError}.
```
Comment (damienwebdev) [inline libs/core/src/storage/server-error/server-error.service.ts:8]: Link needs update.

File: libs/core/src/storage/server-error/server-error.service.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffErrorServerStorageService implements DaffPersistenceService {
```
Comment (damienwebdev) [inline libs/core/src/storage/server-error/server-error.service.ts:13]: `DaffServerErrorStorageService`

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/core/src/storage/server-error/server-error.ts (hunk 1 above)
Comment (griest024) [reply libs/core/src/storage/server-error/server-error.ts]: changed

File: libs/core/src/storage/server-error/server-error.service.ts (hunk 2 above)
Comment (griest024) [reply libs/core/src/storage/server-error/server-error.service.ts]: changed

File: libs/core/src/storage/server-error/server-error.service.ts (hunk 3 above)
Comment (griest024) [reply libs/core/src/storage/server-error/server-error.service.ts]: not sure what you mean but I changed the name here as well

File: libs/core/src/storage/server-error/server-error.service.ts (hunk 4 above)
Comment (griest024) [reply libs/core/src/storage/server-error/server-error.service.ts]: changed

**Review by damienwebdev — APPROVED** [review]

**Pull request merged**